src/utils/parse-accept.ts

```diff
@@ -0,0 +1,86 @@
+export interface Accept {
+  type: string
+  params: Record<string, string>
+  q: number
+}
+
+/**
+ * Parse an Accept header into an array of objects with type, parameters, and quality score.
+ * @param acceptHeader The Accept header string
+ * @returns An array of parsed Accept values
+ */
+export const parseAccept = (acceptHeader: string): Accept[] => {
+  if (!acceptHeader) {
+    return []
+  }
+
+  const acceptValues = acceptHeader.split(',').map((value, index) => ({ value, index }))
+
+  return acceptValues
+    .map(parseAcceptValue)
+    .filter((item): item is Accept & { index: number } => Boolean(item))
+    .sort(sortByQualityAndIndex)
+    .map(({ type, params, q }) => ({ type, params, q }))
+}
+
+const parseAcceptValue = ({ value, index }: { value: string; index: number }) => {
+  const parts = value
+    .trim()
+    .split(/;(?=(?:(?:[^"]*"){2})*[^"]*$)/)
+    .map((s) => s.trim())
+  const type = parts[0]
+  if (!type) {
+    return null
+  }
+
+  const params = parseParams(parts.slice(1))
+  const q = parseQuality(params.q)
+
+  return { type, params, q, index }
+}
```

Comment:
I think this implementation will compile the regex object for each `parseAcceptValue` calling.
So I prefer to declare the regex in the root scope.

```suggestion
const parseAcceptValueRegex = /;(?=(?:(?:[^"]*"){2})*[^"]*$)/
const parseAcceptValue = ({ value, index }: { value: string; index: number }) => {
  const parts = value
    .trim()
    .split(parseAcceptValueRegex)
    .map((s) => s.trim())
  const type = parts[0]
  if (!type) {
    return null
  }

  const params = parseParams(parts.slice(1))
  const q = parseQuality(params.q)

  return { type, params, q, index }
}
```